mlrun/datastore/azure_blob.py

```diff
@@ -53,36 +56,65 @@ def get_storage_options(self):
             connection_string=self._get_secret_or_env(
                 "AZURE_STORAGE_CONNECTION_STRING"
             ),
+            tenant_id=self._get_secret_or_env("AZURE_STORAGE_TENANT_ID"),
+            client_id=self._get_secret_or_env("AZURE_STORAGE_CLIENT_ID"),
+            client_secret=self._get_secret_or_env("AZURE_STORAGE_CLIENT_SECRET"),
+            sas_token=self._get_secret_or_env("AZURE_STORAGE_SAS_TOKEN"),
         )
 
     def upload(self, key, src_path):
-        # Need to strip leading / from key
-        blob_client = self.bsc.get_blob_client(container=self.endpoint, blob=key[1:])
-        with open(src_path, "rb") as data:
-            blob_client.upload_blob(data, overwrite=True)
+        if self.bsc:
+            # Need to strip leading / from key
+            blob_client = self.bsc.get_blob_client(container=self.endpoint, blob=key[1:])
+            with open(src_path, "rb") as data:
+                blob_client.upload_blob(data, overwrite=True)
+        else:
+            remote_path = f"{self.endpoint}{key}"
+            self._filesystem.put_file(src_path, remote_path)
 
     def get(self, key, size=None, offset=0):
-        blob_client = self.bsc.get_blob_client(container=self.endpoint, blob=key[1:])
-        size = size if size else None
-        return blob_client.download_blob(offset, size).readall()
+        if self.bsc:
+            blob_client = self.bsc.get_blob_client(container=self.endpoint, blob=key[1:])
+            size = size if size else None
+            return blob_client.download_blob(offset, size).readall()
+        else:
+            blob = self._filesystem.cat_file(key, start=0)
```

Comment:
Why always use `start=0`? You're ignoring the `offset` and `size` parameters here.

Reply:
self._filesystem.cat_file() calculates the mlrun's size parameter, instead of taking it explicitly.  That being the case, the concern would be that if start is allowed to be anything other than zero, we could create reading overlapping fragments.

I could update adlfs to accept size in a future release to accept size and set that to length.  Thoughts?

Reply:
Yes, I see that `cat_file` indeed accepts `start` and `end` rather than a `size`. It does feel a bit awkward to do a translation just for `adlfs` to reverse the calculation. In this context, adding an option to `adlfs` to accept size will make this better (though not compliant with the `fsspec` default interface).
Still, I'm not sure I follow the overlap concern - why would this translation result in overlaps? Are you concerned about a `size` that exceeds object size?

Reply:
Size exceeding object_size would be a concern.  Another would be multiple calls, where start is zero, then non-zero.

If I add a size parameter to cat_file, I would handle it in kwargs

Reply:
I'm probably misunderstanding your concern here - it's legitimate for a client to generate reads over overlapping regions by specifying overlapping `offset`/`size` parameters. Usually that wouldn't be the case, as you'd use a fixed `size` and loop with `offset = prev_offset + size` until you reach EOF, but it's definitely not the intent to block other usages.

Reply:
I'll pass offset in to start, since it looks like that's a requirement to pass the integration tests.